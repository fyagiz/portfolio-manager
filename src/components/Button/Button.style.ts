import { StyleSheet } from "react-native";
import { STYLE } from "../../utils/constants";

export default StyleSheet.create({
  container: {
    alignSelf: "flex-start",
    margin: STYLE.margin,
    padding: STYLE.padding,
    borderRadius: STYLE.borderRadius,
    maxWidth: STYLE.maxWidth,
  },
});
