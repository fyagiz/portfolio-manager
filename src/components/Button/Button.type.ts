import { GestureResponderEvent } from "react-native";

export type ButtonPropsType = {
  text: string;
  onPress?: (event: GestureResponderEvent) => void;
  testOnly_pressed?: boolean;
};
