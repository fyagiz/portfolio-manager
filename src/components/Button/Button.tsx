import { Pressable, Text } from "react-native";
import { ButtonPropsType } from "./Button.type";
import styles from "./Button.style";
import { Colors } from "../../utils/Colors";

const Button = (props: ButtonPropsType) => {
  const { text, onPress, testOnly_pressed } = props;
  const { buttonColor, pressedButtonColor } = Colors;
  return (
    <Pressable
      style={({ pressed }) => [
        styles.container,
        {
          backgroundColor: pressed ? pressedButtonColor : buttonColor,
        },
      ]}
      onPress={onPress}
      testID="pressable"
      testOnly_pressed={testOnly_pressed}
    >
      <Text testID="buttonText">{text}</Text>
    </Pressable>
  );
};

export default Button;
