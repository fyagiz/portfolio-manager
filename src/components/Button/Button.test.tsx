import { fireEvent, render } from "@testing-library/react-native";
import Button from "./Button";

/*
1- Button should be rendered
*/

describe("Button Component Unit Tests", () => {
  test("should render correctly", () => {
    const buttonComponent = render(<Button text="Test Button" />);
    expect(buttonComponent).toMatchSnapshot();
  });

  test("should show button text", () => {
    const buttonComponent = render(<Button text="Test Button" />);
    const buttonChildren = buttonComponent.getByTestId("buttonText").children;
    const buttonName = buttonChildren.join("");
    expect(buttonName).toMatch("Test Button");
  });

  test("should be pressed and run the onPress function", () => {
    const mockFunction = jest.fn();
    const investmentCardComponent = render(<Button text="Test Button" onPress={mockFunction} testOnly_pressed />);
    const pressable = investmentCardComponent.getByTestId("pressable");
    fireEvent(pressable, "click");
    expect(mockFunction).toBeCalled();
  });
});
