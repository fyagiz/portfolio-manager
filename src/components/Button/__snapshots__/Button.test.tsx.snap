// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Unit Tests should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignSelf": "flex-start",
        "borderRadius": 4,
        "margin": 4,
        "maxWidth": 128,
        "padding": 4,
      },
      {
        "backgroundColor": "#F3CC79",
      },
      undefined,
    ]
  }
  testID="pressable"
>
  <Text
    testID="buttonText"
  >
    Test Button
  </Text>
</View>
`;
